pkg/realizer/workload/realizer.go

```diff
@@ -66,7 +67,21 @@ func (r *realizer) Realize(ctx context.Context, resourceRealizer ResourceRealize
 			}
 		}
 
-		realizedResources = append(realizedResources, generateRealizedResource(resource, template, stampedObject, out, previousResources))
+		realizedResource := generateRealizedResource(resource, template, stampedObject, out, previousResources)
+
+		conditionManagerBuilder := conditions.NewConditionManager
+		conditionManager := conditionManagerBuilder(v1alpha1.ResourceReady, getPreviousResourceConditions(resource.Name, previousResources))
```

Comment:
Not sure if there's supposed to be a pattern here; can probably be inlined without loss of any context?

```suggestion
		conditionManager := conditions.NewConditionManager(v1alpha1.ResourceReady, getPreviousResourceConditions(resource.Name, previousResources))
```